addons/l10n_ke/models/account_tax.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, api
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    l10n_ke_item_code_id = fields.Many2one(
+        'l10n.ke.item.code',
+        string='KRA Item Code',
+        help='KRA code that describes a tax rate or exemption on specific products or services.',
+        compute='_compute_l10n_ke_item_code_id',
+        domain="[('tax_rate', 'in', ['C','E'])] if amount == 0 else [('tax_rate', '=', 'B') if amount == 8 else ('tax_rate', '=', False)]",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends('amount')
+    def _compute_l10n_ke_item_code_id(self):
```

Comment:
I would rather use an onchange in this case (just not sure if we should expand the compute thing to that as well, because it is not really a compute anymore), so it won't trigger by accident and won't the domain already do it.